cmd/app.go

```diff
@@ -28,18 +29,39 @@ var (
 )
 
 func init() {
+	subcommands := [][]string{
+		{"list-images", "list images used by cloud-provider-kind"},
+	}
+
 	flag.IntVar(&flagV, "v", 2, "Verbosity level")
 	flag.BoolVar(&enableLogDump, "enable-log-dumping", false, "store logs to a temporal directory or to the directory specified using the logs-dir flag")
 	flag.StringVar(&logDumpDir, "logs-dir", "", "store logs to the specified directory")
 	flag.BoolVar(&enableLBPortMapping, "enable-lb-port-mapping", false, "enable port-mapping on the load balancer ports")
 
 	flag.Usage = func() {
-		fmt.Fprint(os.Stderr, "Usage: cloud-provider-kind [options]\n\n")
+		fmt.Fprint(os.Stderr, "Usage: cloud-provider-kind [options] <subcommand>\n\n")
+		fmt.Fprintln(os.Stderr, "Options:")
 		flag.PrintDefaults()
+		fmt.Fprint(os.Stderr, "\n")
+		fmt.Fprintln(os.Stderr, "Subcommands:")
+		printSubcommands(subcommands)
 	}
 }
 
 func Main() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "You must specify at least one option or flag.")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	switch os.Args[1] {
+	case "list-images":
+		listImages()
+		return
+	default:
+	}
+
```

Comment:
```suggestion
	// Parse subcommands if exist
	if len(os.Args) > 1 {
		switch os.Args[1] {
		case "list-images":
			listImages()
			return
		default:
		}
	}
```

Reply:
also after flag.VisitAll, line70 add

```
	// don't allow subcommands after flags
	if len(flag.Args()) > 0 {
		flag.Usage()
		os.Exit(1)
	}
```

Reply:
I misunderstood the usage of cloud-provider-kind. Thank you for pointing it out!